pkg/kubelet/server/remotecommand/httpstream.go

```diff
@@ -375,3 +429,34 @@ func handleResizeEvents(stream io.Reader, channel chan<- term.Size) {
 		channel <- size
 	}
 }
+
+func v1WriteErrorFunc(stream io.WriteCloser) func(msg string, rc *int) error {
+	return func(msg string, rc *int) error {
+		_, err := stream.Write([]byte(msg))
+		return err
+	}
+}
+
+// V4ExitErrorMessage objects are marshaled as json in the error channel.
+type V4ExitErrorMessage struct {
+	// Message is the actual human readable text error message.
+	Message string `json:"message"`
```

Comment:
And the fields should be set appropriately - we should have a way to send back exit code, probably as a detail